packages/forms/src/UIForm/UIForm.component.js

```diff
@@ -0,0 +1,177 @@
+import React, { PropTypes } from 'react';
+import { merge } from 'talend-json-schema-form-core';
+
+import { validateValue, validateAll } from './utils/validation';
+import Widget from './Widget';
+
+const TRIGGER_AFTER = 'after';
+
+export default class UIForm extends React.Component {
+	constructor(props) {
+		super(props);
+		const { jsonSchema, uiSchema } = props;
+		this.state = {
+			mergedSchema: merge(jsonSchema, uiSchema),
+		};
+		console.log(this.state.mergedSchema)
+
+		this.onChange = this.onChange.bind(this);
+		this.onTrigger = this.onTrigger.bind(this);
+		this.submit = this.submit.bind(this);
+	}
+
+	/**
+	 * Update the state with the new schema.
+	 * @param jsonSchema
+	 * @param uiSchema
+	 */
+	componentWillReceiveProps({ jsonSchema, uiSchema }) {
+		if (!jsonSchema || !uiSchema) {
+			return;
+		}
+		this.setState({
+			mergedSchema: merge(jsonSchema, uiSchema),
+		});
+	}
+
+	/**
+	 * Triggers the onTrigger and onChange if needed
+	 * - onChange : at each field change
+	 * - onTrigger : when schema.trigger : ['after']
+	 * @param event The event that triggered the callback
+	 * @param schema The field schema
+	 * @param value The new value
+	 */
+	onChange(event, schema, value) {
+		const {
+			formName,
+			onChange,
+			properties,
+			customValidation,
+		} = this.props;
+		const error = validateValue(schema, value, properties, customValidation);
+		onChange(formName, schema, value, error);
+
+		const { triggers } = schema;
+		if (triggers && triggers.indexOf(TRIGGER_AFTER) !== -1) {
+			this.onTrigger(event, TRIGGER_AFTER, schema, value, properties);
+		}
+	}
+
+	/**
+	 * Triggers an onTrigger callback that is allowed to modify the form
+	 * @param event The event that triggered the callback
+	 * @param type The type of trigger
+	 * @param schema The field schema
+	 * @param value The field value
+	 */
+	onTrigger(event, type, schema, value) {
+		const { formName, onFormChange, onTrigger, onValidate, properties } = this.props;
+		if (!onTrigger) {
+			return;
+		}
+
+		onTrigger(
+			type,           // type of trigger
+			schema,         // field schema
+			value,          // field value
+			properties,     // current properties values
+		)
+			.then(newForm => onFormChange(
+				formName,
+				newForm.jsonSchema,
+				newForm.uiSchema,
+				newForm.properties,
+				newForm.errors)
+			)
+			.catch(({ errors }) => onValidate(formName, errors));
+	}
+
+	/**
+	 * Triggers submit callback if form is valid
+	 * @param event the submit event
+	 */
+	submit(event) {
+		event.preventDefault();
+		const { mergedSchema } = this.state;
+		const { formName, properties, customValidation } = this.props;
+		const errors = validateAll(mergedSchema, properties, customValidation);
+
+		const isValid = !Object.keys(errors).length;
+		if (isValid) {
+			this.props.onSubmit(event, properties);
+		} else {
+			this.props.onValidateAll(formName, errors);
```

Comment:
Should we validateAll in all cases on submit ?
Validation here is not done on blur but as it may be later it can not validate field when there is some errors & enter key is pressed to submit the form